test/integration/auth/auth.prose.test.ts

```diff
@@ -0,0 +1,373 @@
+            test: async function () {
+              const username = encodeURIComponent(user.username);
+              const password = encodeURIComponent(user.password);
+
+              const url = `${makeConnectionString(
+                this.configuration,
+                username,
+                password
+              )}authMechanism=${mechanism}`;
+
+              const client = this.configuration.newClient(url);
+              const stats = await client.db('test').stats();
+              expect(stats).to.exist;
+              await client.close();
+            }
+          });
+        }
+
+        it(`authenticates ${user.description} using mechanism negotiaton`, {
+          metadata: { requires: { mongodb: '>=3.7.3' } },
+          test: async function () {
+            const options = {
+              auth: {
+                username: user.username,
+                password: user.password
+              },
+              authSource: 'admin'
+            };
+
+            const client = this.configuration.newClient({}, options);
+            const stats = await client.db('test').stats();
+            expect(stats).to.exist;
+            await client.close();
+          }
+        });
+
+        it(`authenticates ${user.description} using mechanism negotiaton and url`, {
+          metadata: { requires: { mongodb: '>=3.7.3' } },
+          test: async function () {
+            const username = encodeURIComponent(user.username);
+            const password = encodeURIComponent(user.password);
+            const url = makeConnectionString(this.configuration, username, password);
+
+            const client = this.configuration.newClient(url);
+            const stats = await client.db('test').stats();
+            expect(stats).to.exist;
+            await client.close();
+          }
+        });
+      }
+
+      /**
+       * Step 2
+       * For a test user supporting both SCRAM-SHA-1 and SCRAM-SHA-256, drivers should verify
+       * that negotation selects SCRAM-SHA-256. This may require monkey patching, manual log
+       * analysis, etc.
+       * todo(NODE-5629): Test passes locally but will fail on CI runs.
+       */
+      it.skip('selects SCRAM-SHA-256 for a user that supports both auth mechanisms', {
+        metadata: { requires: { mongodb: '>=3.7.3' } },
+        test: async function () {
+          const options = {
+            auth: {
+              username: userMap.both.username,
+              password: userMap.both.password
+            },
+            authSource: this.configuration.db
+          };
+
+          const spy = sinon.spy(ScramSHA256.prototype, 'auth');
+
+          const client = this.configuration.newClient({}, options);
+          const stats = await client.db('test').stats();
+          expect(stats).to.exist;
+          expect(spy.called).to.equal(true);
+          sinon.restore();
+          await client.close();
+        }
+      });
```

Comment:
```suggestion
      }).skipReason = '...'
```
can we attach a skip reason? 

We can also use the predicate filter if you'd prefer:

```typescript
{ requires: { predicate: () => 'TODO(..)...' } }
```

Reply:
Added skip reason.